File: Rasha_Toine/Rasha/QLForm/src/ql/visiting/TypeVisitor.java
package ql.visiting;

import ql.ast.type.*;

public interface TypeVisitor<T, U> {
	public T visit(Type type, U ctx);
	public T visit(BooleanType type, U ctx);
	public T visit(DateType type, U ctx);
	public T visit(StringType type, U ctx);
	public T visit(IntegerType type, U ctx);
	public T visit(MoneyType type, U ctx);
	public T visit(DecimalType type, U ctx);
}

File: henk/src/main/scala/ql/visitors/TypeVisitor.scala
package ql.visitors

import grammar._
import ql.models.ast._

class TypeVisitor extends QLBaseVisitor[NodeType] {

  override def visitTypeDecl(ctx: QLParser.TypeDeclContext): NodeType =
    ctx.getText match {
      case "boolean" => BooleanType()
      case "money"   => MoneyType()
      case "integer" => IntegerType()
      case "string"  => StringType()
    }
}
